### ISSUE TITLE
Does the metaLayer cause loss of information ?

### ISSUE THREAD

**Issue by ICDI0906** [rusty1s/pytorch_geometric#526]
## ❓ Questions & Help

Hi~, I have a graph whose node, edge, global attribute size is 1,1,3 separately. I use MetaLayer to model this. And the task is to learn the embedding of every node.so the question is when the size of the attribute is small, Does there occur loss of information? In addition, my graph is sparse!

**Comment by rusty1s** [comment]
I am not sure I understand what you mean by loss of information. Generally speaking, there should be none, as most of the GNN layers are able to preserve local information.

**Comment by ICDI0906** [comment]
The loss of information is that the performance becomes worse when I add the GNN layer, so I think maybe there occur loss of information.

**Comment by rusty1s** [comment]
Well, that can have multiple sources of error. Which GNN layer do you use? Some GNNs like GCN can be understood as low-pass filters, and hence could result in decreasing performance.

**Comment by ICDI0906** [comment]
Thanks for your prompt reply! I use the [MetaLayer](https://pytorch-geometric.readthedocs.io/en/latest/modules/nn.html?) which is proposed in the paper[Relational inductive biases, deep learning, and graph networks].

**Comment by rusty1s** [comment]
So you use the example script provided in the docs to define your MetaLayer? This is just an example, and it may need to be adjusted for your use-case, *e.g.*, you need to pre-add self-loops to your graph, or preserve local information by replacing the call
```python
out = torch.cat([out, u[batch]], dim=1)
```
with
```python
out = torch.cat([x, out, u[batch]], dim=1)
```

**Comment by ICDI0906** [comment]
Yes, I just use the example to have a try. I don't preserve local information as you said. I will adjust my code to have another try. Thanks again!

**Comment by ICDI0906** [comment]
It achieves better performance when I modify the code as you said! It looks like local information is very important！Thank you very much for your suggestion!
